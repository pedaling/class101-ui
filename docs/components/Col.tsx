// @flow
import { Col, Colors } from '@class101/ui';
import React from 'react';

const hiddenStyle = {
  border: `1px solid ${Colors.orange500}`,
  textAlign: 'center',
  backgroundColor: `${Colors.orange200}`,
  backgroundClip: 'content-box',
};

export const Component = ({ lg, sm, smOffset, lgOffset, style, ...restProps }: Props) => (
  <Col lg={lg} sm={sm} smOffset={smOffset} lgOffset={lgOffset} style={{ ...style, ...hiddenStyle }} {...restProps} />
);

export const NoHiddenStyleCol = ({ lg, sm, smOffset, lgOffset, style, ...restProps }: Props) => (
  <Col lg={lg} sm={sm} style={{ ...styleMedia }} {...restProps} />
);
