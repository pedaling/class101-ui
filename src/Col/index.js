import styled from 'styled-components';
import { media } from '../BreakPoints';

const sizeToPercent = (size: number) => (((size || 12) / 12) * 100);

const Col = styled.div`
  position: relative;
  min-height: 1px;
  padding-right: 12px;
  padding-left: 12px;
  float: left;
  box-sizing: border-box;

  ${media.sm`
    width: ${props => sizeToPercent(props.sm)}%;
    padding-right: 4px;
    padding-left: 4px;
  `};

  ${media.md`
    width: ${props => sizeToPercent(props.md || props.sm)}%;
  `};

  ${media.lg`
    width: ${props => sizeToPercent(props.lg || props.md || props.sm)}%;
  `};
`;

export default Col;
