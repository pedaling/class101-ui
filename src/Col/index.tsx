import styled from 'styled-components';

import { media } from '../BreakPoints';
import { HTMLDivProps } from '../interfaces/props';

const sizeToPercent = (size?: number) => ((size || 12) / 12) * 100;

interface ColProps extends HTMLDivProps {
  sm?: number;
  md?: number;
  lg?: number;
  smOffset?: number;
  mdOffset?: number;
  lgOffset?: number;
}

const Col = styled.div<ColProps>`
  position: relative;
  min-height: 1px;
  padding-right: 12px;
  padding-left: 12px;
  box-sizing: border-box;
  ${media.sm<ColProps>`
    width: ${props => sizeToPercent(props.sm)}%;
    padding-right: 4px;
    padding-left: 4px;
    ${props =>
      props.smOffset &&
      `
        margin-left: ${sizeToPercent(props.smOffset)}%;
      `}
  `}

  ${media.md`
    width: ${props => sizeToPercent(props.md || props.sm)}%;
    ${props =>
      (props.mdOffset || props.smOffset) &&
      `
        margin-left: ${sizeToPercent(props.mdOffset || props.smOffset)}%;
      `}
  `}

  ${media.lg`
    width: ${props => sizeToPercent(props.lg || props.md || props.sm)}%;
    ${props =>
      (props.lgOffset || props.mdOffset || props.smOffset) &&
      `
        margin-left: ${sizeToPercent(props.lgOffset || props.mdOffset || props.smOffset)}%;
      `}
  `};
`;

export default Col;
