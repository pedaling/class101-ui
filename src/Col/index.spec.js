import Col from './';

describe('Col', () => {
  it('값이 존재한다.', () => {
    expect(Col).toBeTruthy();
  });

  it('스냅샷과 일치한다..', () => {
    expect(Col).toMatchSnapshot();
  });
});
