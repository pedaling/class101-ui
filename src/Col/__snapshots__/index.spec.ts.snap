// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col 스냅샷과 일치한다. 1`] = `
Object {
  "$$typeof": Symbol(react.forward_ref),
  "attrs": Array [],
  "componentStyle": ComponentStyle {
    "componentId": "sc-bdVaJa",
    "isStatic": false,
    "rules": Array [
      "
  position: relative;
  min-height: 1px;
  padding-right: 12px;
  padding-left: 12px;
  box-sizing: border-box;
  ",
      "
      @media ",
      "(max-width: 1023px)",
      " {
        ",
      "
    flex-basis: ",
      [Function],
      "%;
    padding-right: 4px;
    padding-left: 4px;
    ",
      [Function],
      "
  ",
      ";
      }
    ",
      "

  ",
      "
      @media ",
      " {
        ",
      "
    flex-basis: ",
      [Function],
      "%;
    ",
      [Function],
      "
  ",
      ";
      }
    ",
      "

  ",
      "
      @media ",
      "(min-width: 1024px)",
      " {
        ",
      "
    flex-basis: ",
      [Function],
      "%;
    ",
      [Function],
      "
  ",
      ";
      }
    ",
      ";
",
    ],
  },
  "displayName": "styled.div",
  "foldedComponentIds": Array [],
  "render": [Function],
  "styledComponentId": "sc-bdVaJa",
  "target": "div",
  "toString": [Function],
  "warnTooManyClasses": [Function],
  "withComponent": [Function],
}
`;
